packages/ra-ui-materialui/src/layout/Menu.tsx

```diff
@@ -89,12 +80,9 @@ const Menu: FC<MenuProps> = props => {
                         leftIcon={
                             resource.icon ? <resource.icon /> : <DefaultIcon />
                         }
-                        onClick={onMenuClick}
```

Comment:
I refactored the `<MenuItemLink>` to let it pull the props it needs (rather than expecting them to be pushed by the `<Sidebar>`). 

This greatly simplifies the creation of custom menus.

Reply:
Indeed :+1: